### DISCUSSION TITLE
Setof record error

### DISCUSSION THREAD

**Discussion by Shohzod97** [supabase/supabase#16075] [Questions] [answered]
I'm trying to get all the rows of a table through a database function.
 This [PostGIS: Geo queries guide](https://supabase.com/docs/guides/database/extensions/postgis) has an example request that starts like this:

```
create or replace function nearby_restaurants(lat float, long float)
returns setof record
...
```

It creates a database function without error.
But when you execute it, it gives:

Unhandled Exception: PostgrestException(message: a column definition list is required for functions returning "record", code: 42601, details: Bad Request, hint: null)

Most recently it all worked, but now it gives this error.

**Comment by steve-chavez** [comment, marked as answer]
@Shohzod97 Let's track this on https://github.com/supabase/postgrest-js/issues/453.

The solution could be a bug fix or an update to the guide.
